### DISCUSSION TITLE
ICustomization for DateOnly causes a hang.

### DISCUSSION THREAD

**Discussion by grosch** [AutoFixture/AutoFixture#1435] [Q&A]
### Describe the Bug

When I used the below fixture customization to handle DateOnly properties, the build/create call hangs forever.

### Scenario

```csharp
public sealed class DateOnlyFixtureCustomization : ICustomization {
    public void Customize(IFixture fixture) => fixture.Customize<DateOnly>(x => x.FromFactory<DateTime>(DateOnly.FromDateTime));
}

public static class FixtureFactory
{
    public static IFixture Create() {
        var fixture = new Fixture().Customize(new DateOnlyFixtureCustomization());

        fixture.Behaviors
            .OfType<ThrowingRecursionBehavior>()
            .ToList()
            .ForEach(x => fixture.Behaviors.Remove(x));

        fixture.Behaviors.Add(new OmitOnRecursionBehavior(2));

        return fixture;
    }
}
```

Then I simply use it like so:

```csharp
var client = FixtureFactory.Create().Build<Client>()
    .With(...)
    .Create();
```

### Expected Behavior

Assign a DateOnly value to the appropriate properties.

### Tasks

I'm using 4.18.0, and from googling it looks like the above is how I'm supposed to make this work

**Comment by aivascu** [comment]
@grosch are you sure the issue is due to the `DateOnly` customization? I see you've added for some reason the `OmitOnRecursionBehavior` which is not required for the `DateOnly` type as it does not have any recursive members, nor any writable properties for that matter.
Could you provide some insight into the `Client` class you're trying to create?
I've tried to reproduce the bug from the code sample, and I can confirm that there is no bug there.
Moving this thread to Discussions.
